### PR TITLE
Drop cache_object protocol support

### PR DESCRIPTION
**Pull request by eduard-bagdasaryan**
Removing this non-standard protocol (already mentioned as deprecated
in Squid sources) helps eliminate duplication and simplifies the
existing error-prone forwarding logic (causing CVEs).

Separate commits replace cache_object URLs sent by squidclient and
cachemgr.cgi tools with URLs using an http scheme.

### CODE REVIEW

File: src/cf.data.pre
Comment (yadij) [inline src/cf.data.pre:1071]: Now that the scheme does not matter we can make this a faster path-only check.
```suggestion
DEFAULT: manager urlpath_regex +i ^/squid-internal-mgr/
```

**Review by yadij — CHANGES_REQUESTED** [review]
The release notes should be for v7 now. Since that file does not yet exist please just drop those chunks or wait on the merge until after v6.0.1 branching adds the necessary release-7.sgml file in a few days.

File: src/cf.data.pre
Comment (rousskov) [reply src/cf.data.pre:1071]: Please do _not_ change the type of the `manager` ACL in this PR, especially for optimization reasons. Doing so will break existing Squid configurations that explicitly adjust the `manager` ACL. Such configurations are problematic, and we are working on prohibiting such adjustments, but that ban should be done (differently) in a separate PR (which is nearly ready for the official review).

**Comment by rousskov** [comment]
> The release notes should be for v7 now. Since that file does not yet exist please just drop those chunks or wait on the merge until after v6.0.1 branching adds the necessary release-7.sgml file in a few days.

While I disagree with nearly every assertion in the above statement, I am OK with removing release notes from this PR if doing so allows us to merge this PR before v6 is branched. @eduard-bagdasaryan, please do that. We can always restore the deleted text later as needed, of course.

File: src/cf.data.pre
Comment (eduard-bagdasaryan) [reply src/cf.data.pre:1071]: > a separate PR (which is nearly ready for the official review)

Yes, it would be better to avoid these changes, since that PR should cover this area.

**Comment by eduard-bagdasaryan** [comment]
>  drop those chunks

Done.

**Comment by yadij** [comment]
> > The release notes should be for v7 now. Since that file does not yet exist please just drop those chunks or wait on the merge until after v6.0.1 branching adds the necessary release-7.sgml file in a few days.
> 
> While I disagree with nearly every assertion in the above statement, I am OK with removing release notes from this PR if doing so allows us to merge this PR before v6 is branched.

Why are you in such a rush to change this?

**Comment by rousskov** [comment]
> Why are you in such a rush to change this?

I am not in a rush. I believe that removing cache_object from future v6 will simplify backporting future master/v7 changes (that do rely on cache_object not being there), and it is easier to merge a ready PR now than to backport it later, but I do not insist on an immediate merge of this PR (or any other PR that is ready for merging now). If you insist on a delay, we will merge after v6 is branched. Do you insist on a delay?

File: src/cf.data.pre
Comment (yadij) [reply src/cf.data.pre:1071]: Is this other PR ready or in yet?
I would definitely like to reduce the security problems the current manager ACL definition could lead to.

**Review by yadij — COMMENTED** [review]
Just the ACL type discussion to resolve. Then this can go into v7.

**Review by yadij — CHANGES_REQUESTED** [review]
Actually, I should now be requesting release notes documentation. This is a major feature change - making the v7 Squid no longer compatible with squidclient or cachemgr.cgi older than v5.8, and with third-party tools using the cache_object scheme to directly access reports.

File: src/cf.data.pre
Comment (eduard-bagdasaryan) [reply src/cf.data.pre:1071]: This PR is passing or internal review - it is not official yet.

**Comment by eduard-bagdasaryan** [comment]
> release notes documentation

Ok, I'll add them into [release-7.sgml.in](https://github.com/squid-cache/squid/blob/master/doc/release-notes/release-7.sgml.in).

File: src/cf.data.pre
Comment (rousskov) [reply src/cf.data.pre:1071]: Please note that the `manager` ACL type should not be changed in _this_ PR regardless of that type (doomed) future. 

> I would definitely like to reduce the security problems the current manager ACL definition could lead to.

Same here, but neither the current type, nor the type suggested in this change request can solve those problems. Please resolve this change request. Hopefully, we will soon post an official PR that properly solves those problems, once and for all, but even if we fail to post that "proper fix" PR soon, any type change for this ACL is outside this PR scope and needs a dedicated PR (for the reasons already [given](https://github.com/squid-cache/squid/pull/1250#discussion_r1093314706)).

File: doc/release-notes/release-7.sgml.in
Comment (rousskov) [inline doc/release-notes/release-7.sgml.in:31]: What we are describing is not really a new feature. It is more like a removal of an old feature, but there is no such section in the template AFAICT. FWIW, I am not against adding a new section in this PR if others want it (here and in template.sgml?). If others can do that instead, that would be even better because I find it difficult to work within this document organization.

File: doc/release-notes/release-7.sgml.in
Comment (rousskov) [inline doc/release-notes/release-7.sgml.in:39]: I tried to polish this a little. If anybody does not like the current wording, please just edit the file to your liking. I do not insist on any specific wording here.

**Comment by rousskov** [comment]
@yadij, please move this PR forward. The release notes updates you have [requested](https://github.com/squid-cache/squid/pull/1250#pullrequestreview-1323436993) have been added and polished. There are no outstanding code change requests AFAICT. This `review-2` PR has been waiting for your review for more than two months.

**Review by yadij — DISMISSED** [review]

**Review by rousskov — APPROVED** [review]
Approving after resolving a small merge conflict in release notes: We now have two items in the "Cache Manager changes" section of those release notes. That section was added in recent master/v7 commit e373298.

**Comment by rousskov** [comment]
@eduard-bagdasaryan, please fix [build errors](https://github.com/squid-cache/squid/actions/runs/5394004062/jobs/9794558168?pr=1250) that were apparently introduced by the recent merge from master.

**Review by rousskov — APPROVED** [review]

**Pull request closed without merging**
